### PR TITLE
add & enable eslint

### PR DESCRIPTION
**Pull request by brodybits**
mostly extracted from PR #102; one or more upcoming PRs will be based off of this one

~~It is an open question whether this PR should be merged or be superseded by another eslint PR.~~

### CODE REVIEW

**Comment by brodybits** [comment]
I would appreciate a quick review of this PR & PR #107.

I would like to merge both of these PRs if there are no criticisms in the next few days. Thanks!

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+env:
+  browser: true
+  commonjs: true
+  node: true
+# FUTURE TBD:
+# extends: 'eslint:recommended'
+parserOptions:
+  ecmaVersion: 6
```
Comment (karfau) [inline .eslintrc.yml:8]: Question: Is this related to the ES6 syntax topic?

Comment (brodybits) [reply .eslintrc.yml:8]: Yes and no, good question.

In general I wanted to enable the lowest ES version possible with the existing code base when starting to add eslint. I had wanted to do `ecmaVersion: 5`, it worked with 100% with `lib` but not with `test`, so I went with `ecmaVersion: 6`.

Considering that PR #107 *does* seem to solve the problem with enforcing the older ES version in `lib`, I am wondering if we need to keep an outdated ecmaVersion number here.

Comment (karfau) [reply .eslintrc.yml:8]: I'm not familiar with configuring eslint, but I think being explicit is often better then relying on implicit things. Less "magic".

So if there is not a good reason to remove the option I would leave it as is.

Comment (brodybits) [reply .eslintrc.yml]: I just put it back to `ecmaVersion: 11` with `es2020: true` in the env options above, according to the output of `npx eslint --init`.

My rationale is that I would not object to ES2020 code in tests, and we need PR #107 with eslint-plugin-es5 to solve the problem with older browsers. But I can go back to `ecmaVersion: 6` again.

Comment (karfau) [reply .eslintrc.yml]: Perfect!
